### DISCUSSION TITLE
Welcome to the TechExcellence Community!

### DISCUSSION THREAD

**Discussion by valentinajemuovic** [techexcellenceio/techexcellence#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by valentinajemuovic** [comment]
To introduce myself - my name is Valentina Cupać, and I'm a Founder and Technical Coach @ Optivem I had worked as a software developer for several years, mainly in enterprise software development (.NET and Java). I really cared about code quality. The turning points point for me was discovering TDD and Hexagonal Architecture - and from then onwards, I never looked backward. 

The biggest challenge I faced in companies was that university programs don't really teach best practices, and employees in companies don't have an opportunity to learn best practices - aside from their own self-education. So we end up just multiplying the spaghetti code. The result is a vicious cycle - many senior developers who have several years of experience in making software work, but not in making maintainable software, and no one to teach the juniors how to code well... 

This was one of the reasons why I became a Technical Coach, to help companies build teams with high-quality foundations.

> "There is no trade-off of quality vs. speed in software... Low quality means low speed... The only way to go fast is to go well." - Uncle Bob

I created this community so that together, we can make a difference in software engineering professionalism across the world. It's about pushing forward the idea that quality and speed go hand-in-hand, there is no tradeoff.

**Comment by ythirion** [comment]
Hi all, I am Yoan Thirion
I am an Agile Technical Coach passionate about agility, craftsmanship and liberated organizations.
I became addicted to agility from my first reading of the manifesto and am firmly convinced that the only way to work is by considering people as individuals in their own right and no longer as "resources".

Since then, I try to help as much as I can teams by training them on practices, techniques that help them deliver high quality software and learn continuously (the true agility = continuous learning)

**Reply by valentinajemuovic** [reply]
@ythirion thanks for the intro. The following is a nice formulation "only way to work is by considering people as individuals in their own right" -> what impact do you see in shifting the thinking from thinking about resources to seeing people as individuals?

**Reply by ythirion** [reply]
It is not an easy one ☺️
it has a huge impact on organization (for the good) but people often don’t feel it.
Often the only guidance or only measure of success is project driven centric : I have delivered something at one point that matches the business needs (once again at one point in time) and that’s it. We don’t care about evolutions, maintainability of the system and so on…
When you work on the dev mindset and culture, developers tend to challenge stuff, try to understand the value of doing or not doing stuff. If the culture is open, people will be open to criticize the system and change it.  but it is not the case in every company…
We can have positive impact on people as tech coaches but not every companies are ready to have software craftsmen as developers

**Comment by PCreations** [comment]
Hi there :) I'm Pierre, a french Craftsman & Formator/Coach
I'm very fond of TDD / Hexagonal architecture / DDD. Since I discovered this world many years ago I try to share my knowledge, mostly on LinkedIn and on the Wealcome Slack, a french community about software craftsmanship.

**Reply by valentinajemuovic** [reply]
@PCreations thanks for the intro, esp. since you're well-known on LinkedIn, I see your posts almost every day. 

I'm interested to hear more about your story in TDD / Hexagonal architecture / DDD - how did you discover it? Did you find out by yourself or you saw it in practice?

Also thanks for recommending Wealcome Slack (cc @mica16) - it's an amazing community.

**Reply by PCreations** [reply]
I've always been passionate about architecture. I first discovered unit testing in 2013 during an internship. Then, from one thing to another, I discoered DDD and all the related things. In 2016 I read Clean Architecture, I found it waaay to complicated at the time, I was struggling because I was coding in Javascript, without classes, without types, so I needed to "translate" from OO to a more functional way, and on the frontend by the way ! This make me discover hexagonal architecture, TDD, and it tooks me one year and a half full time to put it all together. Since that, I keep learning, practicing, and trying to share it along the way :)

**Comment by diogokobbi** [comment]
Hello every one, my name is Diogo, i'm a brazilian software developer, and i'm here to learn about software engineering.
Best regards

**Comment by daltondiaz** [comment]
My name is Dalton and I'm a software developer for several years working with Java (main), PHP, Javascript, Typescript and Go and now the things that I want learning more is code with quality, best practices, patterns (SOLID and cia) and architecture (Clean Arch, Hexagonal) and good Tests (TDD, BDD and etc.). After many years working we see this items above very important to delivery new features fast and well, the software is always change if you want scale fast you need a very good architecture.

So i want contribute and learn more!

**Comment by irwansyahwii** [comment]
Hi! My name is Irwansyah from Jakarta, Indonesia. I am a practitioner of OOP, Design Patterns, SOLID Principles and DDD. I used to practiced the wrong way of TDD and I stopped since it is so messed up. I am planning to migrate to pure functional programming but I found it hard to switch the mindset and learned all the required basic knowledge of pure functional programming.

**Comment by mikewojtyna** [comment]
Hey everyone, nice to meet you. I'm Mike Wojtyna.

I'm a Java software developer with passion for creating great products. Domain-driven design & Test-driven development are some of my favorite tools. I'm also a professional consultant and trainer.

Recently I've created an interactive training program to help you become a professional software developer (https://howtobe.pro/).

**Comment by CromixPT** [comment]
Hello all,

My name is Tiago, from Lisbon, Portugal. I'm a Software Engineer currently focusing on .NET stack.

I spend two thirds of my day thinking I'm an imposter, and the rest learning(discount the sleeping). I'm passionate about building scalable and maintainable software, but I always feel I have much to learn.

Joining the community to learn and also to contribute.

**Reply by ngrahovac** [reply]
Love how you described your day - I can relate! 😂
